packages/foam-vscode/src/features/create-from-template.ts

```diff
@@ -11,54 +11,122 @@ import { FoamFeature } from '../types';
 import { TextEncoder } from 'util';
 import { focusNote } from '../utils';
 
-const templatesDir = `${workspace.workspaceFolders[0].uri.path}/.foam/templates`;
+const templatesDir = `${workspace.workspaceFolders[0].uri.fsPath}/.foam/templates`;
+const firstTemplateContent = `# Your new template!
+
+Templates are inserted the same ways snippets are.
+This means you get access to [all these good features!](https://code.visualstudio.com/docs/editor/userdefinedsnippets#_snippet-syntax).
+For example, check out some tabstops in the below list:
+
+- $1
+- $2
+- $3
+
+To create a note from this template, run the 'Foam: Create new note from template' command.
+`;
 
 async function getTemplates(): Promise<string[]> {
   const templates = await workspace.findFiles('.foam/templates/**.md');
-  // parse title, not whole file!
-  return templates.map(template => path.basename(template.path));
+  return templates.map(template => path.basename(template.fsPath));
+}
+
+async function offerToCreateTemplate(): Promise<void> {
+  const response = await window.showQuickPick(['Yes', 'No'], {
+    placeHolder:
+      'No templates available. Would you like to create one instead?',
+  });
+  if (response === 'Yes') {
+    commands.executeCommand('foam-vscode.create-new-template');
+    return;
+  }
+}
+
+async function createNoteFromTemplate(): Promise<void> {
+  const templates = await getTemplates();
+  if (templates.length === 0) {
+    return offerToCreateTemplate();
+  }
+  const activeFile = window.activeTextEditor?.document?.fileName;
+  const currentDir =
+    activeFile !== undefined
+      ? path.dirname(activeFile)
+      : workspace.workspaceFolders[0].uri.fsPath;
```

Comment:
same here, ideally we would:
- always use `URI`
- if we can't use the `URI`, we use `path`
- if we really can't use `path`, use `fsPath`

if we use `URI` than it will be the `DataStore` to take care of the platform specific behavior.
if we use `fsPath` the rest of that code needs to be platform-safe.

Reply:
Just so we're talking about the same `URI`, do you mean a Foam `URI` and not a `vscode.Uri`?

Reply:
Either one, they are fully compatible. but generally speaking let's use the Foam one so that we keep data in our abstractions as much as possible